### PR TITLE
Consistent span usage in api

### LINKED ISSUE #541: Change space.Info Min/Max Time to Span
In the search request we represent a time range as a `nano.Span`. For consistency sake, `api.SpaceInfo` should do the same.

### PR DESCRIPTION
**Pull request by mattnibs**
Make the api consistent by leveraging nano.Span when expressing time
ranges.

Note: I didn't change the info file to a span. This would require
a migration for older installations. I figure this isn't worth the
effort as at some point we'll probably change the way determine the
time span of a space.

closes #541

### CODE REVIEW

File: zqd/space/space.go [hunk 1]
```diff
@@ -422,6 +422,10 @@ type info struct {
 	MaxTime nano.Ts `json:"max_time"`
 }
 
+func (i info) Span() nano.Span {
+	return nano.NewSpanTs(i.MinTime, i.MaxTime)
+}
+
 // UnsetTimes nils out the cached time range value for the space.
 // XXX For right now this simply deletes the info file as nothing else is stored
 // there. When we get to brimsec/zq#541 the time range should be represented as
```
Comment (henridf) [inline zqd/space/space.go:431]: Looks like this comment should be updated if this isn't being done in #541

File: zqd/ingest/log.go [hunk 2]
```diff
@@ -123,7 +123,9 @@ func ingestLogs(ctx context.Context, pipe *api.JSONPipe, s *space.Space, req api
 		return err
 	}
 	if tailW.r != nil {
-		if err = s.SetTimes(tailW.r.Ts, headW.r.Ts); err != nil {
+		min := nano.Min(tailW.r.Ts, headW.r.Ts)
+		max := nano.Max(tailW.r.Ts, headW.r.Ts)
+		if err = s.SetSpan(nano.NewSpanTs(min, max)); err != nil {
```
Comment (henridf) [inline zqd/ingest/log.go:128]: This span is incorrect as spans are open on the right... I think you need to make it `NewSpanTs(min, max+1)` ?

File: zqd/space/space.go (hunk 1 above)
Comment (mattnibs) [reply zqd/space/space.go:431]: Yep

File: zqd/ingest/log.go (hunk 2 above)
Comment (mattnibs) [reply zqd/ingest/log.go:128]: I think you are right.

Comment (mattnibs) [inline zqd/ingest/log.go:128]:
```suggestion
		if err = s.SetSpan(nano.NewSpanTs(min, max+1)); err != nil {
```

**Comment by henridf** [comment]
Does this have a companion issue over on the brim repo?

**Comment by mattnibs** [comment]
> Does this have a companion issue over on the brim repo?

I will do this

**Pull request merged**